crypto/curve25519/curve25519.c

```diff
@@ -31,7 +31,7 @@
 #pragma warning(disable: 4244)
 #endif
 
-#if defined(__GNUC__)
```

Comment:
Are these changes really needed? This code has been compiled by clang for a long time without complaint...

Reply:
Yeah, so it turns out we're the first ones to compile this code with Clang on Windows 😅

Clang does its best to be compatible with other compilers. On Windows it does its best to be [compatible with MSVC](https://clang.llvm.org/docs/MSVCCompatibility.html), while on Unix it [builds the code in GNU mode](https://clang.llvm.org/compatibility.html#c) by default (whatever that means).

So in fact, if we replaced this line with `#if defined(__GNUC__) || defined(_MSC_VER)`, it would still compile! Also this is the exact reason why a few lines earlier there is `#if defined(_MSC_VER) && !defined(__clang__)`.

So in the end, the answer is "yes", we need these changes because Clang on Windows doesn't define __GNUC__, and then we cannot compile 🤷‍♂️